examples/chainable_router.rs

```diff
@@ -0,0 +1,22 @@
+#[macro_use] extern crate nickel;
+extern crate hyper;
+
+use nickel::{Nickel};
+use nickel::router::chainable_router::{ChainablePath, ChainableHandler};
+
+fn main() {
+    let mut server = Nickel::new();
+    server
+        .route("/")
+        .get(middleware!("get"))
+        .post(middleware!("post"))
+        .put(middleware!("put"))
+        .patch(middleware!("patch"))
+        .delete(middleware!("delete"))
+        .route("/chain1")
```

Comment:
I agree chaining route here is a little weird.  I'll remove that.  That was kind of an afterthought and I didn't really think it through very clearly.